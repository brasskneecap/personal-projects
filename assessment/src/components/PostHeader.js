import ProfilePicture from "./ProfilePicture"

const PostHeader = ({ post, headerType})=> {
    return (
        <div className={headerType}>
            <ProfilePicture imgLocation={post.userOwner.profilePicture}/>
            <div className="titles-container">
                <div className="primary-title">
                    { post.userOwner.name}
                </div>
                <div className="secondary-title">
                    0 Minutes ago
                </div>
            </div>
            <div className="post-options">
                ...
            </div>
        </div>
    )
}

export default PostHeader