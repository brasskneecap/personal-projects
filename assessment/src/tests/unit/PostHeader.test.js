import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import PostHeader from '../../components/PostHeader';

describe('PostHeader', () => {
    const wrapper = new ShallowRenderer()

    let post = {
        id: 1,
        userOwner: {
            name: "Test User",
        },
        content: "Test message", 
        reactions: {
            hypes: 0,
            comments: 0,
            shares: 0,
            views: 0
        },
        comments: []
    }

    wrapper.render(
    <PostHeader post={post} headerType="post-header" />
    );
    
    let componentRendered = wrapper.getRenderOutput();
    
    it('Interaction Snapshot', () => {
        expect(componentRendered).toMatchSnapshot();
    })

    it('Should have post-header className', () => {
        expect(componentRendered.props.className).toBe("post-header")
    })
    
    it('Should have User name in title', () => {
        expect(componentRendered.props.children[1].props.children[0].props.children).toBe("Test User")
    })
})