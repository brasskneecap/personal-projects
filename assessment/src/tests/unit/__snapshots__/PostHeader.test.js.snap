// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostHeader Interaction Snapshot 1`] = `
<div
  className="post-header"
>
  <ProfilePicture />
  <div
    className="titles-container"
  >
    <div
      className="primary-title"
    >
      Test User
    </div>
    <div
      className="secondary-title"
    >
      0 Minutes ago
    </div>
  </div>
  <div
    className="post-options"
  >
    ...
  </div>
</div>
`;
